src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/VaultContentXMLContentCreator.java

```diff
@@ -0,0 +1,366 @@
+    }
+    
+   
+    @Override
+    public void createProperty(String name, Object value)  throws RepositoryException {
+        currentNode.addProperty(name, value);
+    }
+    
+    
+    @Override
+    public void createProperty(String name, Object[] values) throws RepositoryException {
+        currentNode.addProperty(name, values);
+    }
+
+    @Override
+    public void createFileAndResourceNode(String name, InputStream data, String mimeType, long lastModified) throws RepositoryException {
+        this.createNode(name, JcrConstants.NT_FILE, null);
+        this.createNode(JcrConstants.JCR_CONTENT, JcrConstants.NT_RESOURCE, null);
+
+        // ensure sensible last modification date
+        if (lastModified <= 0) {
+            lastModified = System.currentTimeMillis();
+        }
+        this.createProperty(JcrConstants.JCR_MIMETYPE, mimeType);
+        this.createProperty(JcrConstants.JCR_LASTMODIFIED, lastModified);
+        this.createProperty(JcrConstants.JCR_DATA, data);
+    }
+    
+
+    @Override
+    public boolean switchCurrentNode(String subPath, String newNodeType) throws RepositoryException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void createUser(String name, String password, Map<String, Object> extraProperties) throws RepositoryException {
+        try {
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB  .append("create user " + name + " with password " + password + " \n\n")
+                            .append("set properties on authorizable(" + name + ")\n");
+            extraProperties.entrySet().stream().map(this::getSetPropertyString).forEach(repoInitTextSB::append);
+            repoInitTextSB.append("end");
+            
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createGroup(String name, String[] members, Map<String, Object> extraProperties) throws RepositoryException {
+        try {
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB  .append("create group " + name + " \n\n")
+                            .append("add " + String.join(",", members) + " to group " + name)
+                            .append("set properties on authorizable(" + name + ")\n");
+            extraProperties.entrySet().stream().map(this::getSetPropertyString).forEach(repoInitTextSB::append);
+            repoInitTextSB.append("end");
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createAce(String principal, String[] grantedPrivileges, String[] deniedPrivileges, String order) throws RepositoryException {
```

Comment:
what is the order param for? repo-init doesn't allow to specify the order/index of a given entry. if some order is defined in the initial content you would need to order the granted/denied entries accordingly and then pray that it's going to be ok...... alternatively: abort because the order cannot be reflected as repo-init statement.

Reply:
order is to order this entry before an entry in the repository... so this is a problem. We'll have to do work in repo-init first to implement this..